### PR TITLE
Big Sky: AI Powered pattern assembler

### PR DESCRIPTION
**Pull request by artpi**
Related to seGwbA-bigskyai-p2

This introduces the `ai-assembler` flow where pattern assembler decisions are driven by a single prompt.

Zrzut ekranu 2023-09-21 o 16 48 20

## Proposed Changes

* Introduce the `ai-assembler` flow and `site-prompt` step to gather data for the AI call
* Pass the patterns selected from the backend to pattern assembler.

## Testing Instructions

- Get a site
- Run [calypso.live](https://calypso.live?image=registry.a8c.com/calypso/app:build-81588) to use prebuilt version or build from scratch locally
- Go to the `/setup/ai-assembler?siteSlug=YOURSLUG`
- Describe your site
- Submit
- Observe magic

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx [hunk 1]
```diff
@@ -65,13 +65,9 @@ import type { AnyAction } from 'redux';
...
 import './style.scss';
 
-const PatternAssembler = ( {
-	navigation,
-	flow,
-	stepName,
-	noticeOperations,
-	noticeUI,
-}: StepProps & NoticesProps ) => {
+const PatternAssembler = ( props: StepProps & NoticesProps ) => {
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx:68]: Is there any reason to make changes here?

File: client/landing/stepper/declarative-flow/ai-assembler.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+				slug: theme.id,
+				title: theme.name,
+				recipe: {
+					...selectedDesign?.recipe,
+					stylesheet: theme.stylesheet,
+				},
+				design_type: 'assembler',
+			} );
+
+			setIntent( SiteIntent.WithThemeAssembler );
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/ai-assembler.ts:54]: Maybe we can call it `ai-assembler`

File: client/landing/stepper/declarative-flow/internals/steps-repository/ai-site-prompt/index.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+							) }
+						</ActionSection>
+					</form>
+				</div>
+			</>
+		);
+	}
+
+	return (
+		<div className="site-prompt__signup is-woocommerce-install">
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/ai-site-prompt/index.tsx:116]: I guess we don't need `is-woocommerce-install` here 🤔

File: client/landing/stepper/declarative-flow/internals/steps-repository/ai-site-prompt/index.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+			<div className="site-prompt__is-store-address">
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/ai-site-prompt/index.tsx:117]: Same for `site-prompt__is-store-address`. Or even the entire `div` 🤔

File: client/landing/stepper/declarative-flow/internals/steps-repository/ai-site-prompt/index.tsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+				<StepContainer
+					stepName="site-prompt"
+					className={ `is-step-${ intent }` }
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/ai-site-prompt/index.tsx:120]: This class name doesn't seem to be required as well 🤔

File: client/landing/stepper/declarative-flow/ai-assembler.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+import ErrorStep from './internals/steps-repository/error-step';
+import PatternAssembler from './internals/steps-repository/pattern-assembler/lazy';
+import ProcessingStep from './internals/steps-repository/processing-step';
+import { ProcessingResult } from './internals/steps-repository/processing-step/constants';
+import { Flow, ProvidedDependencies } from './internals/types';
+import type { OnboardSelect } from '@automattic/data-stores';
+
+const SiteIntent = Onboard.SiteIntent;
+
+const withAIAssemblerFlow: Flow = {
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/ai-assembler.ts:22]: I guess we can simply call it `aiAssemblerFlow` 😆

**Review by arthur791004 — APPROVED** [review]
Looks good to me! Just left some minor feedback.

**Comment by artpi** [comment]
Thank you @arthur791004  ! 
I am going to merge this to unblock backend work and address your feedback in a follow-up pr.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9038390

Thank you @artpi for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9038390).

**Pull request merged**